src/main/java/com/rabbitmq/jms/client/RMQMessageProducer.java

```diff
@@ -227,7 +227,7 @@ private void internalSend(RMQDestination destination, Message message, int deliv
             deliveryMode = javax.jms.DeliveryMode.NON_PERSISTENT;
 
         /* Normalise message to internal form */
-        RMQMessage rmqMessage = RMQMessage.normalise(message);
+        RMQMessage rmqMessage = RMQMessage.normalise(message, session.getTrustedPackages());
```

Comment:
I think this is an overhead on `send`, but shouldn't hurt.
Although may require additional `trustedPackages` if we are as a producer and "untrusting" consumer at the same time.